pkg/asset/installconfig/kubevirt/validation.go

```diff
@@ -0,0 +1,119 @@
+	allErrs := field.ErrorList{}
+	client, err := clientBuilder()
+	if err != nil {
+		detailedErr := fmt.Errorf("failed to create InfraCluster client with error: %v", err)
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("InfraClusterReachable"), "InfraCluster", detailedErr.Error()))
+
+		return nil, allErrs
+	}
+
+	if _, err := client.ListVirtualMachine(namespace); err != nil {
+		detailedErr := fmt.Errorf("failed to access to InfraCluster with error: %v", err)
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("InfraClusterReachable"), "InfraCluster", detailedErr.Error()))
+
+		return nil, allErrs
+	}
+
+	return client, allErrs
+}
+
+func validateStorageClassExistsInInfraCluster(ctx context.Context, name string, client Client, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if _, err := client.GetStorageClass(ctx, name); err != nil {
+		detailedErr := fmt.Errorf("failed to get storageClass %s from InfraCluster, with error: %v", name, err)
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("StorageClassExistsInInfraCluster"), name, detailedErr.Error()))
+	}
+
+	return allErrs
+}
+
+func validateNetworkAttachmentDefinitionExistsInInfraCluster(ctx context.Context, name string, namespace string, client Client, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if _, err := client.GetNetworkAttachmentDefinition(ctx, name, namespace); err != nil {
+		detailedErr := fmt.Errorf("failed to get network-attachment-definition %s from InfraCluster, with error: %v", name, err)
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("NetworkAttachmentDefinitionExistsInInfraCluster"), name, detailedErr.Error()))
+	}
+
+	return allErrs
+}
+
+func validateIPsInMachineNetworkEntryList(machineNetworkEntryList []types.MachineNetworkEntry, apiVIP string, ingressVIP string, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if err := assertIPInMachineNetworkEntryList(machineNetworkEntryList, apiVIP); err != nil {
+		detailedErr := fmt.Errorf("validation of apiVIP %s in cider %s failed, with error: %v", apiVIP, machineNetworkEntryList, err)
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("IPsInCIDR"), apiVIP, detailedErr.Error()))
+	}
+
+	if err := assertIPInMachineNetworkEntryList(machineNetworkEntryList, ingressVIP); err != nil {
+		detailedErr := fmt.Errorf("validation of ingressVIP %s in cider %s failed, with error: %v", ingressVIP, machineNetworkEntryList, err)
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("IPsInCIDR"), ingressVIP, detailedErr.Error()))
+	}
+
+	return allErrs
+}
+
+func assertIPInMachineNetworkEntryList(machineNetworkEntryList []types.MachineNetworkEntry, ip string) error {
+	ipAddr := net.ParseIP(ip)
+	if ipAddr == nil {
+		return fmt.Errorf("ip %s is not valid IP address", ip)
+	}
+	for _, machineNetworkEntry := range machineNetworkEntryList {
```

Comment:
this code will always break after the first entry

Reply:
I'm not sure I understand this comment
There is list of machineNetworkEntry and need to validate that one of the entries include the provided IP
It can be in the first element in the list (or not)

Reply:
Fixed